pygfx/materials/_background.py

```diff
@@ -136,6 +137,7 @@ def map(self):
 
     @map.setter
     def map(self, map):
+        assert map is None or isinstance(map, Texture)
```

Comment:
At some point, we should configure a linting rule that prevents assertions outside `tests` folders. :) Generic `AssertionError`s are not a great UX.

Reply:
Good point. I admit this is mostly out of lazyness, because they're so mich easier to write than `if ... raise ...` 😛